code/datums/components/projectile_shooter.dm

```diff
@@ -0,0 +1,76 @@
+/*This is the projectile_shooter component, a ripoff of the [/datum/component/sound_player] component used for firing off projectiles when certain signals are triggered.
+	This isn't a replacement for how guns work, this is a simply a modular way to fire bullets from things that don't otherwise fire bullets
+	Usage :
+		target.AddComponent(/datum/component/projectile_shooter, args)
+	Arguments :
+		_projectile_type : What projectile do we shoot?
+		_shot_prob : If we only want to go off some of the time when a signal fires, this is the percent chance for it to fire
+		signal_or_sig_list : Used to register the signal(s) you want to fire a shot when sent
+*/
+/datum/component/projectile_shooter
+	/// What projectile do we shoot?
+	var/projectile_type
+	/// If we only want to go off some of the time when a signal fires, this is the percent chance for it to fire
+	var/shot_prob
+
+/datum/component/projectile_shooter/Initialize(_projectile_type = /obj/projectile/bullet/c10mm, _shot_prob = 100, signal_or_sig_list)
```

Comment:
Please do the `src.projectile_type` trick here instead of naming parameters with prefixed underscores. Underscore names generally tend to represent internal values, a contract that is violated when using named parameters.